### PR TITLE
BLD: fix build failure py3.9.1 on OSX

### PR DESCRIPTION
**Pull request by jbrockmendel**
Ran `brew upgrade` yesterday and ended up bumping from 3.9.0 to 3.9.1, after which I got build failures 

```
    LooseVersion(python_target) < "10.9"
  File "/usr/local/Cellar/python@3.9/3.9.1_2/Frameworks/Python.framework/Versions/3.9/lib/python3.9/distutils/version.py", line 306, in __init__
    self.parse(vstring)
  File "/usr/local/Cellar/python@3.9/3.9.1_2/Frameworks/Python.framework/Versions/3.9/lib/python3.9/distutils/version.py", line 314, in parse
    components = [x for x in self.component_re.split(vstring)
TypeError: expected string or bytes-like object
```

No idea why this just started breaking now.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -435,7 +435,7 @@ def run(self):
             "MACOSX_DEPLOYMENT_TARGET", current_system
         )
         if (
-            LooseVersion(python_target) < "10.9"
+            LooseVersion(str(python_target)) < "10.9"
```
Comment (jreback) [inline setup.py:438]: check if https://docs.python.org/3/library/sysconfig.html#module-sysconfig returns a bytestring (e.g. `get_config_vars()`)

weird

Comment (jbrockmendel) [reply setup.py:438]: int 11

Comment (jreback) [reply setup.py:438]: hmm that's also weird

**Comment by jreback** [comment]
can you add a whatsnew note (1.2.1) as should backport this

**Comment by jbrockmendel** [comment]
whatsnew added + green

**Comment by jreback** [comment]
@meeseeksdev backport 1.2.x

**Pull request merged**